builtin/providers/google/resource_compute_health_check.go

```diff
@@ -0,0 +1,519 @@
+							Default:  "/",
+						},
+					},
+				},
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"self_link": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"timeout_sec": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  5,
+			},
+
+			"unhealthy_threshold": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  2,
+			},
+		},
+	}
+}
+
+func resourceComputeHealthCheckCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	// Build the parameter
+	hchk := &compute.HealthCheck{
+		Name: d.Get("name").(string),
+	}
+	// Optional things
+	if v, ok := d.GetOk("description"); ok {
+		hchk.Description = v.(string)
+	}
+	if v, ok := d.GetOk("check_interval_sec"); ok {
+		hchk.CheckIntervalSec = int64(v.(int))
+	}
+	if v, ok := d.GetOk("healthy_threshold"); ok {
+		hchk.HealthyThreshold = int64(v.(int))
+	}
+	if v, ok := d.GetOk("timeout_sec"); ok {
+		hchk.TimeoutSec = int64(v.(int))
+	}
+	if v, ok := d.GetOk("unhealthy_threshold"); ok {
+		hchk.UnhealthyThreshold = int64(v.(int))
+	}
+	if v, ok := d.GetOk("type"); ok {
+		hchk.Type = v.(string)
+	}
+	if v, ok := d.GetOk("tcp_health_check"); ok {
+		if hchk.Type != "TCP" {
+			return fmt.Errorf("TCP health check declared but type is listed as %s", hchk.Type)
+		}
+		tcpcheck := v.([]interface{})[0].(map[string]interface{})
+		tcpHealthCheck := &compute.TCPHealthCheck{}
+		if val, ok := tcpcheck["port"]; ok {
+			tcpHealthCheck.Port = int64(val.(int))
+		}
+		if val, ok := tcpcheck["proxy_header"]; ok {
+			tcpHealthCheck.ProxyHeader = val.(string)
+		}
+		if val, ok := tcpcheck["request"]; ok {
+			tcpHealthCheck.Request = val.(string)
+		}
+		if val, ok := tcpcheck["response"]; ok {
+			tcpHealthCheck.Response = val.(string)
+		}
+		hchk.TcpHealthCheck = tcpHealthCheck
+	}
+
+	if v, ok := d.GetOk("ssl_health_check"); ok {
+		if hchk.Type != "SSL" {
+			return fmt.Errorf("SSL health check declared but type is listed as %s", hchk.Type)
+		}
+		sslcheck := v.([]interface{})[0].(map[string]interface{})
+		sslHealthCheck := &compute.SSLHealthCheck{}
+		if val, ok := sslcheck["port"]; ok {
+			sslHealthCheck.Port = int64(val.(int))
+		}
+		if val, ok := sslcheck["proxy_header"]; ok {
+			sslHealthCheck.ProxyHeader = val.(string)
+		}
+		if val, ok := sslcheck["request"]; ok {
+			sslHealthCheck.Request = val.(string)
+		}
+		if val, ok := sslcheck["response"]; ok {
+			sslHealthCheck.Response = val.(string)
+		}
+		hchk.SslHealthCheck = sslHealthCheck
+	}
+
+	if v, ok := d.GetOk("http_health_check"); ok {
+		if hchk.Type != "HTTP" {
+			return fmt.Errorf("HTTP health check declared but type is listed as %s", hchk.Type)
+		}
+		httpcheck := v.([]interface{})[0].(map[string]interface{})
```

Comment:
I don't totally understand this comment, can you give me an example configuration that could cause this issue?